### ISSUE TITLE
Custom logo branding not working

### ISSUE THREAD

**Issue by EDIflyer** [filebrowser/filebrowser#2039]
**Description**
Trying to make use of custom branding

**Expected behaviour**
Site name change and icon change

**What is happening instead?**
Name changes OK but no icon change

**Additional context**
Running in Docker container
Have tried Ctrl-F5, private browsing and fresh browser to ensure not a cache issue
Have read issues #1122 and #907 which partially deal with this issue

**How to reproduce?**
Setup filebrowser in docker container
Map volume to container, e.g., `home/ediflyer/containers/filebrowser/branding:/branding`
Change global settings to change name and to set branding directory path to `/branding` (ie. no trailing slash and matching the mounted volume)
Place the custom `logo.svg` in the branding directory

NB I have not created an `img` subdirectory as it wasn't clear from the documentation if that was required (or if so if all the PNG files there were required).
NB I have not created a `custom.css` as unsure what this should include

No errors noted in filebrowser docker log

**Screenshots**
![image](https://user-images.githubusercontent.com/13610277/179097471-c08a3c7b-3b32-4c94-99e6-a5345393f7dd.png)
![image](https://user-images.githubusercontent.com/13610277/179097644-731bfa48-e671-438c-92aa-ec92905e0bf0.png)
![image](https://user-images.githubusercontent.com/13610277/179097598-8af6de2c-3e6e-4152-8305-cc3a9c84aed3.png)

**Comment by ramiresviana** [comment]
When the `Branding directory path` option is set to a directory containing `img/icons` folders, File Browser will use that directory for the icons. A `custom.css` file can be also added to the branding directory allowing to define custom styles.

**Comment by EDIflyer** [comment]
Thanks @ramiresviana.  OK so I've added `img` and `img/icons` directories and moved the svg to the `branding/img` directory as per the layout at https://github.com/filebrowser/filebrowser/tree/master/frontend/public/img.

![image](https://user-images.githubusercontent.com/13610277/179242249-3d0554ab-508d-4533-8957-f3c38bb02258.png)

Ctrl-F5 didn't seem to clear the cache properly but checking in a separate browser this does seem to have finally done the trick 👍

I'm still not sure what the options are to go into a `custom.css` file - is there an example one anywhere?

Also is the documentation somewhere on Github? I'd like to try and tweak it to do a PR. _- ah I've now found it in [the other repo](https://github.com/filebrowser/docs)!_

**Comment by EDIflyer** [comment]
Ooh for favicons I just found the link to the very handy https://realfavicongenerator.net/ mentioned at https://github.com/filebrowser/logo - although interestingly it still doesn't seem to properly replace the icon in the Chrome for Android tab, despite uploading the full selection of icons.

The generated code from that website says to add the following to head:
```
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#2b5797">
<meta name="theme-color" content="#ffffff">
```
but I'm assuming this already exists in Filebrowser?

**Comment by EDIflyer** [comment]
I've just gone through a new setup and am finding this inconsistent again.

I've been sure to double check config file and db paths all set correctly (see recent PRs #2215 and https://github.com/filebrowser/docs/pull/47) however I still find that it is hit or miss as to whether the icons are replaced or not. 

Ctrl-F5 required when changing theme from Light to Dark, made sure `/branding` set in config within app - also tried adding to JSON config file but doesn't seem to change it consistently, even after reloading docker container. Is there some logging somewhere that I can use to check it is properly detecting branding info? the main log doesn't say anything much...
```
11/11/2022 0:49:25 2022/11/11 00:49:25 Using config file: /.filebrowser.json
11/11/2022 0:49:25 2022/11/11 00:49:25 Listening on [::]:80
```

Folder structure seems OK:
![image](https://user-images.githubusercontent.com/13610277/201237403-a7ebf46a-4f55-4731-9d00-b7a85d78608e.png)

PS @ramiresviana would you be able to reopen this issue for me whilst still not fixed?

**Comment by EDIflyer** [comment]
PS interestingly 10 min later on a new device it is showing up OK, yet didn't on private browsing or on a different browser on the original machine when first changed despite Ctrl-F5 and container restart. Is there some sort of background process that takes time to replace branding?

**Comment by EDIflyer** [comment]
So have now found if I setup using the following compose (and create the directory and db file prior to running docker) and no json it seems to work although not immediately on the browser I'm on - is there some sort of caching?  I also set `/branding` before uploading files into it, not sure if that makes a difference?

``` yaml
---
version: '3'
services:
  filebrowser:
    image: filebrowser/filebrowser
    container_name: filebrowser
    user: 1000:1000
    ports:
      - 80/tcp
    volumes:
      - /home/alan/:/srv
      - /home/alan/containers/filebrowser/filebrowser.db:/database.db:rw
      - /home/alan/containers/filebrowser/branding:/branding
    environment:
      - TZ=Europe/London
    restart: unless-stopped
    security_opt:
      - no-new-privileges:true
```

**Comment by louisgrasset** [comment]
Sidenote: you can get the change immediately by doing the following:
1. Devtools
2. Applications
3. Storage
4. Click "Clear site data"

**Comment by EDIflyer** [comment]
Top tip! Thanks @louisgrasset - will add a pull request to the documentation to suggest that.

**Comment by Ghuvieira** [comment]
> **Description** Trying to make use of custom branding
> 
> **Expected behaviour** Site name change and icon change
> 
> **What is happening instead?** Name changes OK but no icon change
> 
> **Additional context** Running in Docker container Have tried Ctrl-F5, private browsing and fresh browser to ensure not a cache issue Have read issues #1122 and #907 which partially deal with this issue
> 
> **How to reproduce?** Setup filebrowser in docker container Map volume to container, e.g., `home/ediflyer/containers/filebrowser/branding:/branding` Change global settings to change name and to set branding directory path to `/branding` (ie. no trailing slash and matching the mounted volume) Place the custom `logo.svg` in the branding directory
> 
> NB I have not created an `img` subdirectory as it wasn't clear from the documentation if that was required (or if so if all the PNG files there were required). NB I have not created a `custom.css` as unsure what this should include
> 
> No errors noted in filebrowser docker log
> 
> **Screenshots** ![image](https://user-images.githubusercontent.com/13610277/179097471-c08a3c7b-3b32-4c94-99e6-a5345393f7dd.png) ![image](https://user-images.githubusercontent.com/13610277/179097644-731bfa48-e671-438c-92aa-ec92905e0bf0.png) ![image](https://user-images.githubusercontent.com/13610277/179097598-8af6de2c-3e6e-4152-8305-cc3a9c84aed3.png)

the correct "Branding directory path" is ".\branding" not "/branding".
it worked for me.

**Comment by EDIflyer** [comment]
@Ghuvieira weird - other paths on that page (e.g. path for users) seem to use a forward slash and with the above `/branding` it is working OK for me - not sure why this one would need a backslash unless it has been coded differently?

**Comment by Ghuvieira** [comment]
@EDIflyer really the other settings work with the forward slash, but I spent 2 weeks trying to find out why my logo wasn't showing up, everything was configured correctly, but it didn't work, so I tried to put a backslash and it worked, I tried so many things, I don't even know how I came up with the idea of putting a backslash there.
![image](https://github.com/filebrowser/filebrowser/assets/49819052/d420ebf5-52ad-4d16-aa84-8e88b06a8786)
![image](https://github.com/filebrowser/filebrowser/assets/49819052/f1e4e83b-1190-47b6-b1e1-d70025859dc3)

**Comment by EDIflyer** [comment]
Interesting! The weird thing is that the customisation is working for me with the forward slash! Guess we need one of the devs to explain but glad it works for you now!

**Comment by Overlord852** [comment]
I figured out a way that worked for me.
In The filebrowser.json File i have the line ""root": "/srv"".
when i did a podman inspect of the filebrowser instance i saw that "/srv" was bound to the shared Volume for filebrowser:

"Type": "bind",
                    "Source": "/srv/dev-disk-by-uuid-xxxxxxx-xxx-xxxx-xxxx-xxxxxxxxxxx",
                    "Destination": "/srv/",
                    "Driver": "",
                    "Mode": "",
                    "Options": [
                         "rbind"
                    ],
                    "RW": true,
                    "Propagation": "rprivate"

So i created a branding directory on that volume with the directorys as in the documentation told and in Feilbrowser Setting i put the following Path:
/srv/branding
and it worked fine.

**Comment by Avenelink** [comment]
Hello, well i'm new to docker... and im not an IT ^^ (and english is not my native laguage)

i have filebrowser installed in my synology and working.

i have theses files :

docker
---branding
------img
---------logo.svg
---------icons
---config
------filbrowser.db

in container manager filebrowser settings i have : 

/docker/filebrother/config -> /config (rw)
/myfolder -> /data (rw)
/docker/filebrowser/branding -> /branding (rw)

and in filebrowser (working correctly), in settings, branding directory path is : 

.\branding

BUT I STILL DON'T HAVE MY OWN LOGO :(

i have tried on another computer and its not working also...

Thanks for your help

**Comment by RedaBousserhane** [comment]
![image](https://github.com/user-attachments/assets/8bb53b46-e0b0-489e-a714-2053875a67c5)
![image](https://github.com/user-attachments/assets/29061e0e-d247-42c6-893f-751c8e6ece63)
![image](https://github.com/user-attachments/assets/f6f4c57b-16c0-4c98-bcce-6c3e5a2141e8)

**Comment by JackKay404** [comment]
I had all the same problems as mentioned here and eventually got it to work. 

For me there were several problems, first was my image file was a png and not a proper svg. Also the name of the file should be "logo.svg" or it wont't work. Second, for the favicon in the browser tab bar, you need to have 32x32 and 16x16 size .png images in order to use the icon in the browser. The website in the docs only generates a 48x48 which presumably doesn't fit. This website worked for me: https://favicon.io/favicon-converter/

This was a massive pain which could have been made easy with about two lines of explanation in the documentation.

**Comment by EDIflyer** [comment]
@JackKay404 I wonder if that site has changed what it generates as it worked OK for me before but that was a couple of years ago! It might be worth you popping in a PR for a documentation change?

**Comment by yugaardiyan** [comment]
I've tried it but it still hasn't worked, I installed it in casaos. please help 
My directory is /DATA/AppData/filebrowser/branding

**Comment by ianhundere** [comment]
> Second, for the favicon in the browser tab bar, you need to have 32x32 and 16x16 size .png images in order to use the icon in the browser. The website in the docs only generates a 48x48 which presumably doesn't fit. This website worked for me: https://favicon.io/favicon-converter/

perfect / thank you / 🙇

**Comment by defencedog** [comment]
For me still the solution to change logo isn't working!
The location `./branding` & `/branding` both works! I am using a [custom css](https://github.com/jniggemann/filebrowser-css) which changes icon according to file extension & it is working, thus, both syntax are OK but `.\branding` is WRONG

The logo however is not read. I have made a proper SVG using AI tool 560x560 (default logo ha similar dimension)
I used 
```
/branding
../img
....logo.svg
```
**Not working**
